web/src/components/widgets/Secrets.tsx

```diff
@@ -97,29 +141,54 @@ const SecretRow = ({ secret, index, setSecrets }: SecretRowProps) => {
   )
 }
 
-const AddSecretRow = ({ setSecrets }: { setSecrets: Setter<Secret[]> }) => {
-  const [inputs, setInputs] = createSignal<Secret>({ name: '', value: '' })
+const AddSecretRow = ({ setSecrets, repoID }: AddSecretRowProps) => {
+  const [inputs, setInputs] = createSignal<Secret>({ key: '', value: '' })
 
-  const addSecret = () => {
-    setSecrets((secrets) => [...secrets, inputs()])
-    setInputs({ name: '', value: '' })
+  const addSecret = async () => {
+    const response = await httpClient.setSecret({
+      repoID,
+      key: inputs().key,
+      value: inputs().value,
+    })
+    if (!('error' in response)) {
+      setSecrets((secrets) => [...secrets, inputs()])
+      setInputs({ key: '', value: '' })
+    }
   }
 
   return (
-    <SecretTableRow isEditing={() => true} inputs={inputs} setInputs={setInputs} type="add">
-      <Button disabled={!inputs().name || !inputs().value} onClick={addSecret}>
+    <SecretTableRow
+      isEditing={() => true}
+      revealSecret={async () => false}
+      inputs={inputs}
+      setInputs={setInputs}
+      type="add"
+    >
+      <Button disabled={!inputs().key || !inputs().value} onClick={addSecret}>
         Add
       </Button>
     </SecretTableRow>
   )
 }
 
-const Secrets = () => {
-  const [secrets, setSecrets] = createSignal<Secret[]>([
-    { name: 'API_KEY', value: '123abc' },
-    { name: 'DB_PASSWORD', value: 'password123' },
-    { name: 'SECRET_TOKEN', value: 'token456' },
-  ])
+const Secrets = ({ repoID }: SecretsProps) => {
+  const [secrets, setSecrets] = createSignal<Secret[]>([])
+
+  onMount(() => {
```

Comment:
Im not sure it works if we change a repoID in the props, perhaps we should use createEffect here